fri/test.py

```diff
@@ -36,9 +37,12 @@ def execute():
 
   return response.text
 
-  # ********
-  # To Download call the url .../download/test?fetch=<filename>&apikey=xyz 
+# or 
+def download():
+  url = "http://127.0.0.1:5000/download/test?fetch=f1.txt&apikey=xyz"
+  webbrowser.open(url)

```

Comment:
You are opening a web browser in the call to the download() method. I think you should just receive the HTTP Response stream and save it as a file. Check how this is handled in CONTROL-CORE Mediator:

def ctl(dir=None):

This may be more complex than just opening the browser.

Python has the means to save the HTTP response streams to enable the download() options without relying on an external webbrowser.